source/istio_virtualservice_test.go

```diff
@@ -1634,3 +1640,115 @@ func (c fakeVirtualServiceConfig) Config() *networkingv1alpha3.VirtualService {
 		Spec: vs,
 	}
 }
+
+func TestVirtualServiceSourceGetGateway(t *testing.T) {
+	type fields struct {
+		virtualServiceSource *virtualServiceSource
+	}
+	type args struct {
+		ctx            context.Context
+		gatewayStr     string
+		virtualService *networkingv1alpha3.VirtualService
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		want    *networkingv1alpha3.Gateway
+		wantErr assert.ErrorAssertionFunc
```

Comment:
A better pattern is to have `expectedError string`. If empty string, assert there is no error. Otherwise, assert the error exists and has the expected string. See registry/dynamodb_test.go in #3648 .

Reply:
On it!

Reply:
lmk how you feel on the changes